### PR TITLE
enforce lowercase providerID RG to match cloud-provider-azure

### LINKED ISSUE #2533: providerId in lowerCase does not match the providerId on the nodes
/kind bug

**What steps did you take and what happened:**
CAPI does several operations based on providerIds to determine the list of VMs in AKS. I have only tested it with AKS 

example:

providerID format present on the node:
```
azure:///subscriptions/<subscriptionuid>/resourceGroups/rg-node/providers/Microsoft.Compute/virtualMachineScaleSets/aks-shared001-123456-vmss/virtualMachines/1789
```
ProviderID format passed to AMMP and later to MP
```
azure:///subscriptions/<subscriptionuid>/resourcegroups/rg-node/providers/microsoft.compute/virtualmachinescalesets/aks-shared001-123456-vmss/virtualmachines/1789

```

This can cause issues when using full providerID to find nodes. Which we do after the bugfix https://github.com/kubernetes-sigs/cluster-api/pull/6971 
Any solution should also address https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1585 where resource groups were required to be in lower case.

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind bug

This PR updates the appropriate VM reconcilers so that they produce a providerID that corresponds with how cloud-provider-azure formulates the Azure node instance name.

See:

- https://github.com/kubernetes-sigs/cloud-provider-azure/blob/88c9b89611e7c1fcbd39266928cce8406eb0e728/pkg/provider/azure_vmss.go#L372
- https://github.com/kubernetes-sigs/cloud-provider-azure/blob/88c9b89611e7c1fcbd39266928cce8406eb0e728/pkg/provider/azure_standard.go#L537

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2533

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
enforce lowercase providerID RG to match cloud-provider-azure
```

### CODE REVIEW

File: azure/services/scalesets/scalesets.go [hunk 1]
```diff
@@ -104,7 +105,11 @@ func (s *Service) Reconcile(ctx context.Context) (retErr error) {
 		}
 
 		if fetchedVMSS != nil {
-			s.Scope.SetProviderID(azure.ProviderIDPrefix + fetchedVMSS.ID)
+			providerID, err := azureutil.ConvertResourceGroupNameToLower(azure.ProviderIDPrefix + fetchedVMSS.ID)
```
Comment (CecileRobertMichon) [inline azure/services/scalesets/scalesets.go:108]: can we add a comment to explain why this is needed (to match the cloud-provider format)?

File: azure/services/virtualmachines/virtualmachines.go [hunk 2]
```diff
@@ -98,7 +99,11 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
-		s.Scope.SetProviderID(azure.ProviderIDPrefix + infraVM.ID)
+		providerID, err := azureutil.ConvertResourceGroupNameToLower(azure.ProviderIDPrefix + infraVM.ID)
```
Comment (CecileRobertMichon) [inline azure/services/virtualmachines/virtualmachines.go:102]: ditto

**Comment by jackfrancis** [comment]
/hold

We want a bit more manual tests before we merge this. Stretch goal: add automated test coverage of nodeRef <--> Machine/MachinePool affinity.

**Comment by jackfrancis** [comment]
/hold cancel

this looks good in my comprehensive MachinePool and MachineDeployment tests across both Linux and Windows (we already have this functional coverage in automated E2E but I wanted to be extra sure)

/test pull-cluster-api-provider-azure-e2e-optional

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**